### PR TITLE
Learning typescript

### PR DESCRIPTION
**Pull request by maltron**
I took the liberty in refactored the code into more Typescript. Nothing major, but I spent sometime learning about Patternfly+React using Typescript. 

Hope it helps.

### CODE REVIEW

File: src/app/utils/useDocumentTitle.ts [hunk 1]
```diff
@@ -1,13 +1,10 @@
 import * as React from 'react';
 
-// a custom hook for setting the page title
```
Comment (riccardo-forina) [inline src/app/utils/useDocumentTitle.ts:3]: This comment was kinda useful, I’d keep it

File: src/app/utils/useDocumentTitle.ts [hunk 2]
```diff
@@ -1,13 +1,10 @@
...
-export function useDocumentTitle(title: string) {
+export const useDocumentTitle = (title: string): void =>
```
Comment (riccardo-forina) [inline src/app/utils/useDocumentTitle.ts:3]: We don’t really need a fat arrow here, a plain old function would work as well.

File: src/app/utils/utils.ts [hunk 3]
```diff
@@ -1,8 +1,7 @@
-export function accessibleRouteChangeHandler() {
-  return window.setTimeout(() => {
-    const mainContainer = document.getElementById('primary-app-container');
+export const accessibleRouteChangeHandler = (): number =>
```
Comment (riccardo-forina) [inline src/app/utils/utils.ts:1]: We don’t really need a fat arrow here, a plain old function would work as well.

**Comment by maltron** [comment]
Thanks Ricardo for your inputs. I am just a little confused on 'why' the plain old functions would make more sense as opposed to the fat arrow. Any particular reason ?

**Comment by riccardo-forina** [comment]
I guess it's more a matter of personal preference, but I tend to use arrow functions when chaining "functional" methods like array map or reduce, or when I _need_ to deal with `this` in a reliable way, which in React when not using class based components doesn't happen that much.

Arrow functions are anonymous and, when not in a `{}` block, difficult to debug.

For the specific points where I commented, using an arrow function has no benefit over a plain old function.

For Typescript and React function components in the specific, I tend to use fat arrows because the typing `const CompName: React.FunctionComponent<CompNameProps> = props => ...` automatically injects the `children` prop in the typings and sets the return type so that I can be aware of mistakes while coding. This syntax is more complicated if you want to use a plain old function `function CompName(props: React.PropsWithChildren<CompNameProps>) { ... }`.

**Comment by seanforyou23** [comment]
Closing this one as stale, feel free to reopen if you want to discuss it more!

**Pull request closed without merging**